azurerm/internal/services/securitycenter/resource_arm_security_center_subscription_pricing.go

```diff
@@ -25,7 +22,15 @@ func resourceArmSecurityCenterSubscriptionPricing() *schema.Resource {
 		Delete: resourceArmSecurityCenterSubscriptionPricingDelete,
 
 		Importer: &schema.ResourceImporter{
-			State: schema.ImportStatePassthrough,
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
Unfortunately, `tf import` doesn't work if I don't set the resource type in the resource data schema in the func above.

Reply:
if you don't define the state function, the `Read` func

> Hi @beandrad
> Thanks for this PR, it's off to a good start.
> 
> We try to avoid using multiple API versions for a service wherever possible. I've had a quick scan over the service and I think this should be possible here without too much difficulty. Since the changes you are targeting in the resource are pretty much required for the API bump could you update to just using v3.0? I've put some other comments in below that will also need addressing. We can help you out as needed, and if you're not already joined the community slack, there are instructions for joining in the project [readme](https://github.com/terraform-providers/terraform-provider-azurerm#terraform-provider-for-azure-resource-manager).

Thanks for reviewing! 

You're right, the change to upgrade to v3 is actually trivial; should be fixed now. I created a new commit for this change, do you want me to squash instead.